### ISSUE TITLE
Setting camera bounds to `projection.visibleRegion` is not identity transformation

### ISSUE THREAD

**Issue by fynngodau** [maplibre/maplibre-native#1451]
**Describe the bug**

Triggering a camera update which sets the camera bounds to `projection.visibleRegion` moves the camera up.

**To Reproduce**

To see the effect, I have added a loop that keeps doing this operation on the most basic map from the sample Android app:

```
diff --git a/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/org/maplibre/android/testapp/activity/maplayout/SimpleMapActivity.kt b/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/org/maplibre/android/testapp/activity/maplayout/SimpleMapActivity.kt
index 55f4265c6..9c41edc25 100644
--- a/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/org/maplibre/android/testapp/activity/maplayout/SimpleMapActivity.kt
+++ b/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/org/maplibre/android/testapp/activity/maplayout/SimpleMapActivity.kt
@@ -24,6 +26,14 @@ class SimpleMapActivity : AppCompatActivity() {
                     val styleUrl = styles[0].url
                     maplibreMap.setStyle(Style.Builder().fromUri(styleUrl))
                 }
+
+                maplibreMap.addOnCameraIdleListener {
+                    maplibreMap.moveCamera(CameraUpdateFactory.newLatLngBounds(
+                        maplibreMap.projection.visibleRegion.let {
+                            LatLngBounds.fromLatLngs(listOf(it.farLeft!!, it.farRight!!, it.nearLeft!!, it.nearRight!!))
+                        }, 0
+                    ))
+                }
             }
         )
     }
```

For full diff including imports, see cf8f6db58c46109e48efdcf5d6bfcb416abbd173 from https://github.com/e-foundation/maplibre-gl-native/tree/demo-wrong-transformation.

With this modification:

* the camera keeps moving upwards
* the camera does wrong transformations if it is pitched or zoomed
* possibly additional conditions trigger wrong transformations

**Expected behavior**

The described transformation should be equivalent to the identity transformation.

**Screenshots**

https://github.com/maplibre/maplibre-native/assets/16943720/e936f4f2-c8a3-4e21-bd34-068955056cf9

**Platform**
* Android
* (other platforms not tested)

**Additional context**
For some reason, one app using microG is pretty much doing the loop described above and we would like this app to work.

**Comment by louwers** [comment]
Any idea what causes this? Does it also happen when the map is full screen?

**Comment by fynngodau** [comment]
No, I don't know the cause. First I suspected it might have to do with different calculations regarding earth curvature, but this seems to have no influence on these calculations (a polygon would render as a square, and we are just specifying the four corner points anyway, not the entire area). Whether it happens with a full screen map, I have not tried, but I suspect that the renderer is indifferent about this.
